lib/view_component/slot.rb

```diff
@@ -12,6 +12,24 @@ def initialize(parent)
       @parent = parent
     end
 
+    def content?
+      if (defined?(@__vc_content) && @__vc_content.present?) ||
+          (defined?(@__vc_content_set_by_with_content) && @__vc_content_set_by_with_content.present?) ||
+          (defined?(@__vc_content_block) && @__vc_content_block.present?)
+        return true
+      end
+
+      return @__vc_component_instance.content? if __vc_component_instance?
```

Comment:
```suggestion
      return false if !__vc_component_instance?

      @__vc_component_instance.content?
```

Can we make sure this always returns a `true` or `false` value and not `nil`?

I think with the above guard clause change, this would have a pretty clear pattern of success/failure cases.